app/src/ui/release-notes/release-notes-dialog.tsx

```diff
@@ -0,0 +1,188 @@
+    endpoint: getDotComAPIEndpoint(),
+    fullName: 'desktop/desktop',
+    fork: false,
+    hash: '',
+  },
+  true
+)
+
+const ReleaseNoteHeaderLeftUri = encodePathAsUrl(
+  __dirname,
+  'static/release-note-header-left.svg'
+)
+const ReleaseNoteHeaderRightUri = encodePathAsUrl(
+  __dirname,
+  'static/release-note-header-right.svg'
+)
+
+interface IReleaseNotesProps {
+  readonly onDismissed: () => void
+  readonly emoji: Map<string, string>
+  readonly newRelease: ReleaseSummary
+}
+
+/**
+ * The dialog to show with details about the newest release
+ */
+export class ReleaseNotes extends React.Component<IReleaseNotesProps, {}> {
+  private onCloseButtonClick = (e: React.MouseEvent<HTMLAnchorElement>) => {
+    if (this.props.onDismissed) {
+      this.props.onDismissed()
+    }
+  }
+
+  private renderCloseButton() {
+    // We're intentionally using <a> here instead of <button> because
+    // we can't prevent chromium from giving it focus when the the dialog
+    // appears. Setting tabindex to -1 doesn't work. This might be a bug,
+    // I don't know and we may want to revisit it at some point but for
+    // now an anchor will have to do.
+    return (
+      <a className="close" onClick={this.onCloseButtonClick}>
+        <Octicon symbol={OcticonSymbol.x} />
+      </a>
+    )
+  }
+
+  private renderList(
+    releaseEntries: ReadonlyArray<ReleaseNote>,
+    header: string
+  ): JSX.Element | null {
+    if (releaseEntries.length === 0) {
+      return null
+    }
+
+    const options = new Array<JSX.Element>()
+
+    for (const [i, entry] of releaseEntries.entries()) {
+      options.push(
+        <li key={i}>
+          <RichText
+            text={entry.message}
+            emoji={this.props.emoji}
+            renderUrlsAsLinks={true}
+            repository={repository}
+          />
+        </li>
+      )
+    }
+
+    return (
+      <div className="section">
+        <p className="header">
+          <strong>{header}</strong>
+        </p>
+        <ul className="entries">{options}</ul>
+      </div>
+    )
+  }
+
+  private drawSingleColumnLayout(release: ReleaseSummary): JSX.Element {
+    return (
+      <div className="container">
+        {this.renderList(release.bugfixes, 'Bugfixes')}
+        {this.renderList(release.enhancements, 'Enhancements')}
+        {this.renderList(release.other, 'Other')}
+      </div>
+    )
+  }
+
+  private drawTwoColumnLayout(release: ReleaseSummary): JSX.Element {
+    return (
+      <div className="container">
+        <div className="column">
+          {this.renderList(release.enhancements, 'Enhancements')}
+          {this.renderList(release.other, 'Other')}
+        </div>
+        <div className="column">
+          {this.renderList(release.bugfixes, 'Bugfixes')}
+        </div>
+      </div>
+    )
+  }
+
+  public render() {
+    const release = this.props.newRelease
+
+    const contents =
+      release.enhancements.length > 0 && release.bugfixes.length > 0
```

Comment:
I would consider structuring this differently. Instead of having two separate functions with duplicate work, perhaps we could write a render function for each column that render conditionally and use a consistent flexbox setup to lay them out. Would this work with the `Other` category of notes? I'm not sure what those are.

Reply:
These rules were written to reflect the guidance here: https://github.com/desktop/desktop/issues/2774#issuecomment-332538406. I'll call out the important section:

> Right now our parsing code on desktop.github.com supports these categories new, fixed, improved, removed, and added. With this we'll group new, added, and improved under Enhancements, fixed under Bug Fixes and removed plus any items we're unable to parse or which lacks a category under Other.
>
> If, after grouping, we only end up with a single group we'll render that group full width while decreasing the max width of the update dialog by 50px.
>
> ![](https://user-images.githubusercontent.com/634063/30918158-0677ed64-a39e-11e7-8fb1-79662de7ed52.png)
>
> If there's more than one group the Enhancements group will always be positioned first, in the first column, followed by Bug fixes in the second column. If there's three groups the third, Other, will be positioned in the first column underneath Enhancements.
>
> Illustrated here by a non-balanced version
>
> ![](https://user-images.githubusercontent.com/634063/30918268-5adfd646-a39e-11e7-9edd-19983736912f.png)
>
> If there'd been any 'Other' items here they'd appear 10px below the last item in the Enhancements category.

I don't really have any strong feels on either approach - I think this was just easier to write at the time based on the info I had.

Reply:
I think that's something we should hold off on as we have figured out what this should look like going forward. Right now, the functions don't really duplicate any actual logic, so I'm fine with having two that look similar for now.

Reply:
I noticed this comment here:

> and removed plus any items we're unable to parse or which lacks a category under `Other`.

Which I read as "map any labels that we don't recognize to `Other`" rather than the rigid parsing we currently have. I pushed 32f5b6a17 which adds a test and updates this parsing behaviour.